category/application/fake/category_service.go

```diff
@@ -0,0 +1,75 @@
+package fake
+
+import (
+	"context"
+
+	"flamingo.me/flamingo/v3/framework/flamingo"
+
+	"flamingo.me/flamingo-commerce/v3/category/domain"
+)
+
+//go:generate go run github.com/go-bindata/go-bindata/v3/go-bindata -nometadata -pkg fake -prefix mock/ mock/
+
+// CategoryService returns category test data
+type CategoryService struct {
+	testDataFiles map[string]string
+	logger        flamingo.Logger
+}
+
+var _ domain.CategoryService = new(CategoryService)
+
+// Inject dependencies
+func (f *CategoryService) Inject(logger flamingo.Logger, config *struct {
+	TestDataFolder string `inject:"config:commerce.category.fakeService.testDataFolder,optional"`
+}) {
+	f.logger = logger.WithField(flamingo.LogKeyModule, "category")
+	if config != nil {
+		if len(config.TestDataFolder) > 0 {
+			f.testDataFiles = RegisterTestData(config.TestDataFolder, f.logger)
+		}
+	}
+}
+
+// Tree returns the tree the given category belongs to
+func (f CategoryService) Tree(_ context.Context, activeCategoryCode string) (domain.Tree, error) {
+	categoryTree := LoadCategoryTree(f.testDataFiles, f.logger)
+	index := findTreeRootIndex(activeCategoryCode, categoryTree)
+	if index == -1 {
+		return nil, domain.ErrNotFound
+	}
+	return categoryTree[index], nil
+}
+
+// Get the category of the given category code
+func (f CategoryService) Get(_ context.Context, categoryCode string) (domain.Category, error) {
+	category := LoadCategory(categoryCode, f.testDataFiles, f.logger)
```

Comment:
Currently feels like an overkill to need separate files for each category since that information is already in the categroyTree.json.. Do you plan on providing more detailed information in the single JSON? otherwise, we could just traverse through the category tree and pick the matching category with all its information from there. Maybe easier for testers to edit just one file.